src/script/script_gui.cpp

```diff
@@ -929,6 +895,76 @@ struct ScriptDebugWindow : public Window {
 		}
 	}
 
+	/**
+	 * Update the scrollbar and scroll position of the log panel.
+	 */
+	void UpdateLogScroll()
+	{
+		this->SetWidgetDisabledState(WID_SCRD_SCROLLBAR, script_debug_company == INVALID_COMPANY);
+		if (script_debug_company == INVALID_COMPANY) return;
+
+		ScriptLogTypes::LogData &log = this->GetLogData();
+
+		int scroll_count = (int)log.size();
+		if (this->vscroll->GetCount() != scroll_count) {
+			this->vscroll->SetCount(scroll_count);
+
+			/* We need a repaint */
+			this->SetWidgetDirty(WID_SCRD_SCROLLBAR);
+		}
+
+		if (log.empty()) return;
+
+		/* Detect when the user scrolls the window. Enable autoscroll when the
```

Comment:
I know you just moved stuff around, but this comment could probably be on one line.